statistics/table.go

```diff
@@ -356,8 +356,60 @@ func (t *Table) GetRowCountByIndexRanges(sc *stmtctx.StatementContext, idxID int
 		return getPseudoRowCountByIndexRanges(sc, indexRanges, float64(t.Count), colsLen)
 	}
 	result, err := idx.getRowCount(sc, indexRanges)
-	result *= idx.getIncreaseFactor(t.Count)
-	return result, errors.Trace(err)
+	if result >= 0 || err != nil {
+		return result * idx.getIncreaseFactor(t.Count), errors.Trace(err)
+	}
+	// cannot use the index stats to estimate, fall back to use the column stats.
+	return t.getIndexRowCountByColumns(sc, idx.Info, indexRanges)
```

Comment:
What if we have index stats but don't have column stats?

Reply:
In this case, both estimations are inaccurate, we cannot know which one is better...